app/controllers/restaurants_controller.rb

```diff
@@ -19,6 +19,7 @@ def new
 
   def create
     @restaurant = Restaurant.new(restaurant_params)
+    @restaurant.user_id = current_user.id
```

Comment:
Here I think you can do `@restaurant = current_user.restaurants.build(restaurant_params)` not sure but put a binding.pry and try that out.